tools/provision.py

```diff
@@ -782,6 +781,8 @@ def provision_pki(target, aws, cert_issuer):
     # Generate a key
     print("Generating a new public/private key pair")
     pub_key = target.generate_key()
+    # Fix up the key which is returned with the second to last character incorrect.
+    pub_key = pub_key.replace(b"A=", b"==")
```

Comment:
'=' is a base64 padding character.

Reply:
Thanks for the advice here. I believe I found the bug in the firmware. I'll update this PR shortly.